### ISSUE TITLE
[Bug]: Context menu is unable to highlight the airport bus

### ISSUE THREAD

**Issue by lonelybud** [Yimura/YimMenu#2028]
### Describe the bug

![Screenshot 2023-08-26 024405](https://github.com/YimMenu/YimMenu/assets/139460769/ba407a38-f909-4f75-9638-ee85df182555)

![Screenshot 2023-08-26 024435](https://github.com/YimMenu/YimMenu/assets/139460769/70d3a886-d4bd-4a34-a71e-ef83473d0c87)

![Screenshot 2023-08-26 024424](https://github.com/YimMenu/YimMenu/assets/139460769/12e3551e-93ed-4a1a-9141-85bfa7b5791f)

![Screenshot 2023-08-26 024417](https://github.com/YimMenu/YimMenu/assets/139460769/0092488f-193b-4437-8d4a-95b649980b4f)

### Steps To Reproduce

Spawn a bus
Toggle Context menu in menu > self.
Press X and aim at bus.

### Expected Behavior

Bus should be highlighted by menu.

### Logs and/or screenshots

```
<paste your logs here>
```

### Store

Epic Games

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by T14D3** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/73843330/99fe416e-8008-410c-abbc-bacae061022a)

Can't replicate.

**Comment by lonelybud** [comment]
ok let me try again today. I was in story mode

**Comment by Yimura** [comment]
This is a bug with the way we loop over the entities in the game, sometimes entities don't exist in the list we use.

**Comment by lonelybud** [comment]
today i checked again in story mode. i am not able to replicate this. Dont know why it happened yesterday.

**Comment by lonelybud** [comment]
> This is a bug with the way we loop over the entities in the game, sometimes entities don't exist in the list we use.

this is what you are referring 
https://github.com/YimMenu/YimMenu/blob/5991f0ca292697076628c73db042d3db21b6a956/src/util/entity.cpp#L264

**Comment by alyxme** [comment]
> This is a bug with the way we loop over the entities in the game, sometimes entities don't exist in the list we use.

That is interesting! Sometimes I wonder why I can't select some of players and I thought about that they have some protection from that. So would we have that kind new feature to remove self (and PV) from that list for protection purpose?
